README.md

```diff
@@ -1,2 +1,70 @@
 # pipe-ipc-java
 IPC entry point for local non-http interprocess communication
+
+# Quickstart
+
+The library contains a simple IpcCommandExecutor implementation `EchoExecutor` that writes the given input data to its output stream.
+
+To see this working on a unix computer, take the following steps:
+
+0) required installed software:
+
+* git
+* maven
+* netcat
+
+1) check out the code and build the classes (requires installed git and maven)
+
+```
+$ git clone https://github.com/procilon/pipe-ipc-java
+$ cd pipe-ipc-java
+$ mvn compile
+```
+
+2) run netcat to create the command socket, the input socket and the output socket - since each command is non-stopping command, a new terminal should be used for each command.
```

Comment:
```suggestion
## Creating the communication channels

This library implements the receiving end of the communication. This means that it expects the initiator (in this case we on the commandline) to create and manage the channels for the command JSONs, input and output. In this tutorial we use Unix domain sockets to send messages between different commandline windows.

Open 3 terminal windows: one for sending a command JSON, that tells our executor what we want to do, one for sending an input message and one for receiving the resulting output message.

In each window use netcat to create the respective socket in the tmp directory. This will start a process in each window.
```